appinventor/buildserver/src/com/google/appinventor/buildserver/resources/runtime.scm

```diff
@@ -1427,7 +1427,7 @@
   ;; NOTE: The elements in list-of-strings should be Kawa strings
   ;; but they might not be Java strings, since some (all?) Kawa strings
   ;; are FStrings.  See JavaJoinListOfStrings in components/runtime/utils
-  (JavaJoinListOfStrings:joinStrings list-of-strings separator))
+  (JavaJoinListOfStrings:joinStrings (yail-list-contents list-of-strings) separator))
```

Comment:
I would not change this here as it breaks `get-json-display-representation` defined above. Instead, in your code to generate the YAIL you can call `(yail-list-contents)` so that the `list-of-strings` parameter is correct. Otherwise, when we print a list as a JSON array, the first element is dropped. You can verify this by creating a "make a list" block with two elements and using "Do It" to evaluate it when ShowListsAsJSON in Screen1 is set to true. You will only see the second element printed in this case.

Reply:
Whenever I call (yail-list-contents) in the generator, I get the error:
The operation join with separator cannot accept he arguments: , [["a", "b", "c"]], [","]

Tried so far:
* Renaming "list" to "(yail-list-contents)"
* Wrapping argument1 (the list) with (yail-list-content)
* Replacing argument 1 with (yail-list-content (<list>) '(list) "get list contents") structure

None of them seemed to work. Is there another place I can try to put the yail-list-contents call?

Reply:
One solution that instantly comes to mind is defining a new function in runtime.scm, which would call join-strings with (yail-list-contents) wrapped list.

Is there another approach to do it directly in the JavaScript generator script?

Reply:
That would be fine as well. Define a `yail-join-with-separator` function that calls through to `join-with-separator` and update your generator to call the `yail-` prefixed version.